_docs/technical/protocols/2017-01-13-binary.md

```diff
@@ -559,48 +573,182 @@ Example:
+and used as IDs for wallets and accounts. Addresses without derivation path
+included in the attributes are not meant to contain any money.
+
+#### Address Payload
+
+``` haskell
 data HDAddressPayload = HDAddressPayload !ByteString
     deriving (Eq, Ord, Show, Generic)
 ```
 
 [Source code
-link](https://github.com/input-output-hk/cardano-sl/blob/64bc3ade3555dd4035da3f4bcb15223c9c22f518/core/Pos/Core/Types.hs#L115).
+link](https://github.com/input-output-hk/cardano-sl/blob/524bb4d6d6620bec7e269e33de1154d2b1be5183/core/Pos/Crypto/HD.hs#L50)
+
+`HDAddressPayload` datatype represents a derivation path (`[Word32]`),
+symmetrically encrypted via `ChaChaPoly1305` algorithm. An encryption key may be
+any `ByteString` of 32 bytes, known only to owner of the address.
 
-| Size | Type                      | Value | Description                                             |
-|------|---------------------------|-------|---------------------------------------------------------|
-| 1-2  | TinyVarInt                | n + m | sise of `PubKeyAddress` content                         |
-| n    | Maybe (Word8, \[Word32\]) |       | Empty for nothing and list of Word32 preceded with zero |
-| m    | ByteString                |       | Remaining bytes                                         |
+In current Cardano implementation, encryption key is a 32-byte key derived from
+root public key of wallet using PBKDF2 derivation algorithm with HMAC-SHA512 as
+pseudorandom function, 500 iterations and string `address-hashing` as a salt.
+[Source code link](https://github.com/input-output-hk/cardano-sl/blob/524bb4d6d6620bec7e269e33de1154d2b1be5183/core/Pos/Crypto/HD.hs#L58)
 
-See examples in the next section.
+#### Address Stake Distribution
+
+``` haskell
+-- | Stake distribution associated with an address.
+data AddrStakeDistribution
+    = BootstrapEraDistr
+    -- ^ Stake distribution for bootstrap era.
+    | SingleKeyDistr !StakeholderId
+    -- ^ Stake distribution stating that all stake should go to the given stakeholder.
+    | UnsafeMultiKeyDistr !(Map StakeholderId CoinPortion)
+    -- ^ Stake distribution which gives stake to multiple
+    -- stakeholders. 'CoinPortion' is a portion of an output (output
+    -- has a value, portion of this value is stake). The constructor
+    -- is unsafe because there are some predicates which must hold:
+    --
+    -- • the sum of portions must be @maxBound@ (basically 1);
+    -- • all portions must be positive;
+    -- • there must be at least 2 items, because if there is only one item,
+    -- 'SingleKeyDistr' can be used instead (which is smaller).
+    deriving (Eq, Ord, Show, Generic, Typeable)
+```
+
+[Source code
+link](https://github.com/input-output-hk/cardano-sl/blob/524bb4d6d6620bec7e269e33de1154d2b1be5183/core/Pos/Core/Types.hs#L137)
+
+In Byron era, all addresses should use `BootstrapEraDistr` as stake
+distribution -- which means, all the stake is automatically delegated to
+bootstrap stakeholders.
```

Comment:
You just intrigued a reader, prepared him to explanation of rocket science solution we're going to use soon, and...
just finished here 🐈

Reply:
I added couple of words about Shelley era in the next sentence, but I don't think that this particular file is the right place to describe delegation schemes in detail.